ntoskrnl/io/iomgr/iofunc.c

```diff
@@ -3907,6 +3907,8 @@ NtWriteFile(IN HANDLE FileHandle,
         /* Give the drivers something to understand */
         CapturedByteOffset.u.LowPart = FILE_WRITE_TO_END_OF_FILE;
         CapturedByteOffset.u.HighPart = -1;
+        FileObject->CurrentByteOffset.u.LowPart = FILE_WRITE_TO_END_OF_FILE;
```

Comment:
Certainly not. The CurrentByteOffset of a FileObject always contains the actual offset.